numba/cuda/tests/cudapy/test_intrinsics.py

```diff
@@ -469,25 +477,60 @@ def test_hdiv_scalar(self):
         np.testing.assert_allclose(ary[0], ref)
 
     @skip_unless_cc_53
-    def test_hdiv_scalar2(self):
-        compiled = cuda.jit("void(f2[:], f2, f2)")(simple_hdiv_scalar)
-        ary = np.zeros(1, dtype=np.float16)
+    def test_hdiv(self):
+        compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hdiv_kernel)
         arry1 = np.random.randint(-65504, 65505, size=500).astype(np.float16)
         arry2 = np.random.randint(-65504, 65505, size=500).astype(np.float16)
-        for arg1, arg2 in np.nditer([arry1, arry2]):
-            compiled[1, 1](ary, arg1, arg2)
-            ref = arg1 / arg2
-            np.testing.assert_allclose(ary[0], ref)
+        ary = np.zeros_like(arry1, dtype=np.float16)
+
+        compiled.forall(ary.size)(ary, arry1, arry2)
+        ref = arry1 / arry2
+        np.testing.assert_allclose(ary, ref)
 
     @skip_unless_cc_53
-    def test_hdiv_unique_reg_predicate(self):
-        compiled = cuda.jit("void(f2[:], f2, f2)")(simple_hdiv_unique)
-        ary = np.zeros(1, dtype=np.float16)
-        arg1 = np.float16(3.1415926)
-        arg2 = np.float16(1.57)
-        compiled[1, 1](ary, arg1, arg2)
-        ref = (arg1 / arg2) + (arg2 / arg1)
-        np.testing.assert_allclose(ary[0], ref)
+    def test_hdiv_denormal(self):
+        compiled = cuda.jit("void(f2[:], f2[:], f2[:])")(simple_hdiv_kernel)
+
+        # This test is designed to test divison of fp16 constants that
+        # result in small denormal numbers that should not be further
+        # iterated using the Newton Raphson algorithm used in the hdiv
+        # intrinisc. A smaple of values were extracted from the Nvidia hdiv
+        # test suite
+        arry1 = np.array([0x5b, 0xaf, 0xcb, 0x13b, 0x157, 0x173, 0x18f,
```

Comment:
Why are these values written in hex?

Reply:
That is how the constants were provided in the hdiv test suite.